route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -43,35 +42,47 @@ namespace route_following_plugin
         plugin_discovery_msg_.capability = "strategic_plan/plan_maneuvers";
         pose_sub_ = nh_->subscribe("current_pose", 1, &RouteFollowingPlugin::pose_cb, this);
         twist_sub_ = nh_->subscribe("current_velocity", 1, &RouteFollowingPlugin::twist_cd, this);
-        
-        pnh_->param<double>("minimal_maneuver_duration", mvr_duration_, 16.0);
-        pnh2_->param<double>("config_speed_limit",config_limit);
-        pnh_->param<double>("buffer_time_lanechange",buffer_lanechange_time_);
-        pnh2_->param<double>("guidance/route_end_jerk", jerk_, 1.0);
-        pnh2_->param<std::string>("guidance/lane_change_plugin", lane_change_plugin_);
+
         wml_.reset(new carma_wm::WMListener());
         // set world model point form wm listener
         wm_ = wml_->getWorldModel();
-
+        //set a route callback to update route and calculate maneuver
+        wml_->setRouteCallback(route_set);
+        if(route_set)    
+        {
+            route_cb();
+        }
+        
         discovery_pub_timer_ = pnh_->createTimer(
-            ros::Duration(ros::Rate(10.0)),
-            [this](const auto&) { plugin_discovery_pub_.publish(plugin_discovery_msg_); });
+        ros::Duration(ros::Rate(10.0)),
+        [this](const auto&) { plugin_discovery_pub_.publish(plugin_discovery_msg_); });
     }
+
     void RouteFollowingPlugin::run()
     {
         initialize();
         ros::CARMANodeHandle::spin();
     }
-    bool RouteFollowingPlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp)
-    {        
+
+    void RouteFollowingPlugin::pose_cb(const geometry_msgs::PoseStampedConstPtr& msg)
+    {
+        pose_msg_ = geometry_msgs::PoseStamped(*msg.get());
```

Comment:
Probably out of scope for this PR, but I find it problematic that the strategic plugins subscribe to pose and current speed. Similar to the trajectory planners the arbitrator should provide a consistent state to all strategic plugins such that when the arbitrator compares two plans they both start from the same downtrack distance. Under the current system there can be mismatch between starting states. Could potentially cause issues with switching over to something like platooning.  
@kjrush can prioritize such a change.

Reply:
That's interesting, I actually hadn't considered that possibility, that a new state update could occur in between two plugins generating plans, but that kind of a race condition does make sense. I'd hope the time difference would not be too large (given that the planning itself would hopefully not take long) but I don't have data to confirm or deny that, and if the difference isn't too large the state delta shouldn't be large either. But I can definitely see how that might impact us in edge cases (first plan is at very end of lanelet 1, then we get an update, second plan is entirely within lanelet 2). And if it the time difference is large that's a whole problem unto itself.

I'll have to think on that, but that's a good note. The fix here would just be to push the same kind of state info into PlanManeuver.srv like we have for trajectories (as you mentioned) and then remove those subscribers from the strategic plugin. I do like that flow.

Reply:
Sounds good